process-engine/iam
# ⚡️Add claim check caching

## Changes

1. Implement internal `ClaimCheckCache` for storing the results of a claim check against the authority
    - It can be enabled or disabled through configuration or manually
    - The cache periodically clears out outdated entries
    - The expected config contains the following parameters: 
        - `enabled`: If true, the cache is automatically enabled at startup
        - `cacheLifetimeInSeconds`: The TTL for each cache entry (default is 300 seconds, 0 means the values will never expire)
        - `cleanupIntervalInSeconds`: Interval in which the cache clears out outdated entries (default is 120 seconds, 0 means the cache is never cleared)
    - The config must be stored under `iam:iam_service:cache`
2. IamService uses cached claim check results if available
3. Add `moment` to `dependencies`
4. Add `mocha` and `should` to `devDependencies`
5. Add Unit tests for all components

## Issues

Part of https://github.com/process-engine/process_engine_runtime/issues/358

PR: #11

## How to test the changes

Run the unit tests to verify the integrity of the cache.

For manual testing:
- Use the IamService with an enabled cache
- Make repeated requests for the same claim check
- Note how only the first will be run against the authority, while all the others use the cache
- Also note how the cache gets invalidated within the configured interval
- After a cache value was invalidated, the next request for the same claim will run against the authority again, the result of which is cached again
